chain/client/src/sync/external.rs

```diff
@@ -126,6 +168,29 @@ impl ExternalConnection {
                 }
                 Ok(file_names)
             }
+            ExternalConnection::GCS { gcs_client, bucket, .. } => {
+                let prefix = format!("{}/", directory_path);
+                tracing::debug!(target: "state_sync_dump", shard_id, ?directory_path, "List state parts in GCS");
+                Ok(gcs_client
```

Comment:
I don't understand how `ListRequest` can work with `gcs_client` in the case if no credentials are provided.
Doesn't `gcs_client` expect valid credentials?
How can a node sync if it doesn't provide valid credentials?

Reply:
Ah, indeed, only dump nodes uses that.
Sync nodes assume the files exist.

Reply:
Added comment to this function as well to make sure that somebody doesn't accidentally rely on it for syncing.